deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/CodegenConfig.java

```diff
@@ -1,68 +1,59 @@
-@ConfigRoot(name = CodegenConfig.CODEGEN_TIME_CONFIG_PREFIX, phase = ConfigPhase.BUILD_TIME)
 public class CodegenConfig {
 
     static final String CODEGEN_TIME_CONFIG_PREFIX = "openapi-generator.codegen";
 
     public static final String API_PKG_SUFFIX = ".api";
     public static final String MODEL_PKG_SUFFIX = ".model";
-    public static final String VERBOSE_PROPERTY_NAME = "quarkus." + CODEGEN_TIME_CONFIG_PREFIX + ".verbose";
-    public static final String INPUT_BASE_DIR = "quarkus." + CODEGEN_TIME_CONFIG_PREFIX + ".input-base-dir";
-    public static final String INCLUDE_FILES = "quarkus." + CODEGEN_TIME_CONFIG_PREFIX + ".include";
-    public static final String EXCLUDE_FILES = "quarkus." + CODEGEN_TIME_CONFIG_PREFIX + ".exclude";
-    public static final String VALIDATE_SPEC_PROPERTY_NAME = "quarkus." + CODEGEN_TIME_CONFIG_PREFIX + ".validateSpec";
-    public static final String DEFAULT_SECURITY_SCHEME = "quarkus." + CODEGEN_TIME_CONFIG_PREFIX + ".default.security.scheme";
     // package visibility for unit tests
     static final String BUILD_TIME_GLOBAL_PREFIX_FORMAT = "quarkus." + CODEGEN_TIME_CONFIG_PREFIX + ".%s";
     static final String BUILD_TIME_SPEC_PREFIX_FORMAT = "quarkus." + CODEGEN_TIME_CONFIG_PREFIX + ".spec.%s";
 
     private static final String BASE_PACKAGE_PROP_FORMAT = "%s.base-package";
 
-    public static final String SKIP_FORM_MODEL = "skip-form-model";
-    public static final String MUTINY = "mutiny";
-    public static final String ADDITIONAL_MODEL_TYPE_ANNOTATIONS = "additional-model-type-annotations";
-    public static final String ADDITIONAL_API_TYPE_ANNOTATIONS = "additional-api-type-annotations";
-    public static final String TYPE_MAPPINGS = "type-mappings";
-    public static final String IMPORT_MAPPINGS = "import-mappings";
-    public static final String NORMALIZER = "open-api-normalizer";
+    public static final List<String> SUPPORTED_CONFIGURATIONS = Arrays.stream(ConfigName.values()).map(cn -> cn.name)
+            .collect(Collectors.toList());
+
+    public enum ConfigName {
+        //global configs
+        VERBOSE("verbose"),
+        INPUT_BASE_DIR("input-base-dir"),
+        INCLUDE("include"),
+        EXCLUDE("exclude"),
+        VALIDATE_SPEC("validateSpec"),
+        DEFAULT_SECURITY_SCHEME("default-security-scheme"),
+
+        //spec configs only
+        BASE_PACKAGE("base-package"),
+
+        //global & spec configs
+        SKIP_FORM_MODEL("skip-form-model"),
+        MUTINY("mutiny"),
+        ADDITIONAL_MODEL_TYPE_ANNOTATIONS("additional-model-type-annotations"),
+        ADDITIONAL_API_TYPE_ANNOTATIONS("additional-api-type-annotations"),
+        TYPE_MAPPINGS("type-mappings"),
+        IMPORT_MAPPINGS("import-mappings"),
+        NORMALIZER("open-api-normalizer"),
+        RETURN_RESPONSE("return-response"),
+        ENABLE_SECURITY_GENERATION("enable-security-generation");
+
+        private String name;
+        //TODO: type, default
```

Comment:
TODO?

Reply:
i thought that enum could be enahanced and refactored to contain also default value and type so the whole config loading can be more generic, jsut an idea

Reply:
Sounds like a good idea.
Would you mind opening an issue and removing the comment? if we leave it here as a TODO comment it will end up forgotten forever.

Reply:
removed